src/coreclr/vm/ceeload.cpp

```diff
@@ -1078,17 +1073,13 @@ void Module::SetDebuggerInfoBits(DebuggerAssemblyControlFlags newBits)
     m_dwTransientFlags |= (newBits << DEBUGGER_INFO_SHIFT_PRIV);
 
 #ifdef DEBUGGING_SUPPORTED
-    BOOL setEnC = ((newBits & DACF_ENC_ENABLED) != 0) && IsEditAndContinueCapable();
-
-    // The only way can change Enc is through debugger override.
-    if (setEnC)
+    if (IsEditAndContinueCapable())
     {
-        EnableEditAndContinue();
-    }
-    else
-    {
-        if (!g_pConfig->ForceEnc())
-            DisableEditAndContinue();
+        BOOL setEnC = (newBits & DACF_ENC_ENABLED) != 0 || g_pConfig->ForceEnc() || (g_pConfig->DebugAssembliesModifiable() && CORDisableJITOptimizations(GetDebuggerInfoBits()));
```

Comment:
Note that `IsEditAndContinueCapable` returns false when the JIT optimizations are enabled for module.

It suggests that `CORDisableJITOptimizations` is unnecessary here and this new setting behaves same as the existing ForceEnC switch for all practical purposes. It may be worth capturing this as a comment since it is not obvious by looking at the code.

Reply:
I assumed up to this point that when the ForceEnc env var was set all release and debug built binaries are ENC'able (except system, dynamic, resource and ready to run). If what you implied above that DACF_ALLOW_JIT_OPTS is already set when the Module or ENC module is created, then yes `CORDisableJITOptimizations` will always return the same thing (except something about profiling or instrumentation).  But I may be confusing between the flags set in the Assembly with the ones set in the Module/ENC Module.  I need to dig deeper into this.  This is a lot more messy and complicated than I was hoping.